react/ProductHighlights.tsx

```diff
@@ -1,26 +1,43 @@
-import React, { FC, useContext, ReactNode, useMemo } from 'react'
-import { useProduct, ProductTypes } from 'vtex.product-context'
+import type { FC, ReactNode } from 'react'
+import React, { useContext, useMemo } from 'react'
+import { defineMessages } from 'react-intl'
+import type { ProductTypes } from 'vtex.product-context'
+import { useProduct } from 'vtex.product-context'
 
+import { FILTER_TYPE, QUERY_TYPE } from './constants/EnumTypes'
 import { getSeller } from './modules/seller'
 
-type HighlightType = 'collection' | 'promotion' | 'teaser'
+type HighlightType =
```

Comment:
This type definition will simply resolve to a generic `string`. For example, the following code will not trigger any TS errors:

```ts
const QUERY_TYPE = {
  collection: 'collection',
  promotion: 'promotion',
  teaser: 'teaser',
}

type HighlightType =
  | typeof QUERY_TYPE.collection
  | typeof QUERY_TYPE.promotion
  | typeof QUERY_TYPE.teaser

const a: HighlightType = 'testing'
```

I tend to not like creating this kind of constants, especially in this case, since this is only being used in this file. I suggest simply defining this type as:

```ts
type Highlight = 'collection' | 'promotion' | 'teaser'
``` 

This is easier to read and also to maintain :)

Reply:
@victorhmp

Correction made.